crates/ide-assists/src/handlers/inline_call.rs

```diff
@@ -301,7 +301,18 @@ fn inline(
     params: &[(ast::Pat, Option<ast::Type>, hir::Param)],
     CallInfo { node, arguments, generic_arg_list }: &CallInfo,
 ) -> ast::Expr {
-    let body = fn_body.clone_for_update();
+    let body = if sema.hir_file_for(fn_body.syntax()).is_macro() {
+        cov_mark::hit!(inline_call_defined_in_macro);
+        if let Some(body) = ast::BlockExpr::cast(insert_ws_into(fn_body.syntax().clone())) {
+            body
+        } else {
+            // FIXME(zachs18): I believe this should be unreachable,
+            // since insert_ws_into shouldn't change the kind of the SyntaxNode.
+            fn_body.clone_for_update()
+        }
```

Comment:
```suggestion
        ast::BlockExpr::cast(insert_ws_into(fn_body.syntax().clone()))?
```

Is the `else` block truly unreachable? I.e. will `insert_ws_into` never change the `kind` of the `SyntaxNode`?

Reply:
The suggestion is wrong, since `inline` returns an `ast::Expr`, not an `Option<ast::Expr>`, but my question still stands.

Reply:
`insert_ws_into` should not change the syntax node kind yes, otherwise that is a bug. It's fine to have this unreachable fallback though, assuming there is a bug in `insert_ws_into` it'd be nice to not bring down assists with a panic